src/network/networkd-link.c

```diff
@@ -2385,7 +2384,7 @@ static int link_enter_join_netdev(Link *link) {
         assert(link->network);
         assert(link->state == LINK_STATE_PENDING);
 
-        link_set_state(link, LINK_STATE_ENSLAVING);
+        link_set_state(link, LINK_STATE_CONFIGURING);
```

Comment:
hmm? this should remain ENSLAVING, no?

I mean, you are merging SETTING_ADDRESSES and SETTING_ROUTES, but ENSLAVING should remain the way it is, no?

Reply:
Enslaving is treated as a stage of configuring the link. Any ordering between the sub-stages of configuring should be handled internally. Also, nothing conditionally depends on the value of enslaving and the mapping of this enum to strings called this "configuring" just like SETTING_*

Reply:
not sure i follow... why have LINK_STATE_ENSLAVING at all then?

Reply:
What I was trying to say is that enslaving is just a part of link configuration that is no different from the address / route / neighbor configuration. It could actually be done in parallel with those operations since it requires the bond / bridge to be up. It's not a unique state like COMPLETED / FAILED / INVALID / UNMANAGED / PENDING.